pkg/kubernetes/scripts/policy.js

```diff
@@ -174,6 +177,66 @@
                 return result;
             }
 
+            var sarCache = { };
+
+            function subjectAccessReview(project, user, verb, resource) {
+                var key = (project || "") + ':' + (user ? user.metadata.name : "")+ ':' + verb + ':' + resource;
+                // cached already?
+                if (key in sarCache)
+                    return sarCache[key];
+
+                // pre-fill cache to avoid multiple concurrent requests
+                sarCache[key] = false;
+
+                loader.watch("Projects").then(function() {
+                    // if no project was given, check the first one as an approximation
+                    // TODO: is that sufficient, or should we iterate over all
+                    // projects? (much more expensive)
+                    if (user && !project) {
+                        var projects = select().kind("Project");
+                        for (var p in projects) {
+                            project = projects[p].metadata.name;
+                            break;
+                        }
+                        if (!project)
+                            return false; // the user has no projects
+                    }
+
+                    var request = {
+                        kind: "SubjectAccessReview",
+                        apiVersion: "v1",
+                        namespace: project,
+                        verb: verb,
+                        resource: resource
+                    };
+
+                    methods.post(loader.resolve("subjectaccessreviews"), request)
+                        .then(function(response) {
+                            sarCache[key] = response.allowed;
+                            // tell UI to update only if allowed, as our "no lookup yet" default is false already
+                            if (response.allowed) {
+                                $rootScope.$applyAsync();
```

Comment:
A-ha: binding to promises in templates was [deprecated in Angular 1.2 and removed in 1.3](http://stackoverflow.com/questions/13033118/angular-js-how-to-bind-to-promises), this is also mentioned in the [migration notes](https://docs.angularjs.org/guide/migration#migrating-from-1-2-to-1-3). So apparently the only way that  works now is to actually resolve the promises in the controller and update the property value.

I didn't find any method to have the `ng-show` property re-evaluated after changing the property other than `$applyAsync`, do you have a bright idea about this? But at least with your suggested approach the scope should be more narrow.